docs/native.md

```diff
@@ -0,0 +1,236 @@
+
+Updates the data associated with the current user. Use this to provide updated custom user attributes and metrics, a user ID, and demographic data.
+
+**`Altis.Analytics.record( eventType <string> [, data <object>] )`**
+
+Records an event of `eventType`. This can be any string. The optional data passed in should be an object with either or both an `attributes` object and `metrics` object:
+
+```js
+{
+  attributes: {
+    name: 'value' // <string>
+    // ...
+  },
+  metrics: {
+    name: 1 // <number>
+    // ...
+  },
+}
+```
+
+Those attributes and metrics can be later queried via Elasticsearch.
+
+**`Altis.Analytics.registerAttribute( name <string>, value <string | callback> )`**
+
+Sometimes you may want to record a dynamic attribute value for all events on the page. The `registerAttribute()` allows this. If a function is passed as the value will be evaluated at the time an event recorded.
+
+**`Altis.Analytics.registerMetric( name <string>, value <string | callback> )`**
+
+Similar to `registerAttribute()` above but for metrics.
+
+### Filters
+
+The plugin provides a few hooks for you to control the default endpoint data and attributes recorded with events.
+
+**`altis.analytics.data.endpoint <array>`**
+
+Allows you to provide server-side data used to update the visitors associated endpoint with custom attributes and demographic data. Mostly useful for providing extra data for logged in sessions such as IP based location data.
+
+**`altis.analytics.data.attributes <array>`**
+
+Allows you to provide server-side data recorded for all events on the page. Useful if you need to query records based on the page context.
+
+**`altis.analytics.data.metrics <array>`**
+
+Similar to the `altis.analytics.data.attributes` filter but allows you to pass metrics based on server-side data.
+
+**`altis.analytics.data <array>`**
+
+Filters the entire array passed to the client side.
+
+**`altis.analytics.elasticsearch.url <string>`**
+
+Filters the Elasticsearch server URL.
+
+### Functions
+
+**`Altis\Analytics\Utils\query( array $query, array $params = [] ) : array`**
+
+Queries the analytics data with the Elasticsearch Query DSL provided in `$query`. `$params` will be added as query string parameters to the Elasticsearch request URL.
+
+**`Altis\Analytics\Utils\get_elasticsearch_url() : string`**
+
+Get the Elasticsearch server URL.
+
+**`Altis\Analytics\Utils\milliseconds() : integer`**
+
+Get the current time since the unix epoch in milliseconds.
+
+**`Altis\Analytics\Utils\merge_aggregations( array $current, array $new, string $bucket_type = '' ) : array`**
+
+A utility function for merging aggregations returned by Elasticsearch queries. This is necessary to keep a store of your analytics data as Elasticsearch indexes can be rotated or lost.
+
+## Querying Data
+
+Data can be queried from Elasticsearch, providing a powerful tool for filtering and aggregating records. If you are using Altis requests are automatically signed.
```

Comment:
This plugin / Altis docs is a bit confusing. Can we tweak all these docs to be specific to use in Altis?

Reply:
My bad, copy pasta fail. I'll go through them again and double check